---
title: "Vignette"
author: "Safa-tahar-henni Safia"
date: "29 octobre 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction
In this vignette we will see the main features of the package `PSRank`

Genome-wide association studies (GWASs) are designed to identify genetic susceptibility factors for complex diseases. They are effective approaches for identifying genetic polymorphisms associated with disease risk. Early GWASs of major psychiatric disorders (schizophrenia, bipolar disorder, major depressive disorder, autism spectrum disorders, and attention-deficit hyperactivity disorder) revealed few "hits" that passed the significance thresholds. We are still far from the hoped-for 80% of explained variance due to the genetic heritability expected (Huang et al., 2010). This has contributed to the so-called "Missing heritability" problem which states that unique genetic variations can not explain the vast majority of the heritability of complex diseases. Several explanations for missing heritability have been proposed (Wray et al., 2014) ranging from a larger number of weaker effects polymorphisms, to rarer polymorphisms (with possible greater effects) that would be poorly detected by available genotyping methods (which focus on polymorphisms present in 5% or more of the population) . Overall, it boils down to a relatively low statistical power as heist. Which depends on three factors:

- The size of the cohort.
- The size of the effect.
- The metric used.

The PSRank's package concentrate on the metric used. To recapture missing signal it creates a polygenic's model. For this, the population is separate into two sets of data:

- A learning set where will be estimated associations through linear univariate model. This set of SNPs is going to be sorted. The packages allows to sorte the SNPs according differentes scores. Classically, by P-values or ??. The Lo et al., 2015 's article offer an alternative statistic to the ??2, the I-score: designed to recognize strongly predictive variables rather than highly significant. This score can be used to sort a set of SNPs.
- And a test set where will be calculated a cumulative score, called "polygenic risk score" developed by Purcell et al. and on which we will evaluate the predictivity


To install the package PSRank and load it into R (version 3.4.2):

```{r}
devtools::install_github("SafiaSafa/PSRank")
library(PSRankp)
```

## Load library
This packages needs other library to work:

```{r}
library("pROC")
library("matrixStats")
library("caTools")
library("snpStats")
# library for visualisation of the results
library("ggplot2")
library("tidyverse")
```

## Load data

The data are in the `data/` directory. They are produce by simulation.
The data files are PLINK Binary files (BED/BIM/FAM).
`data/sim9d.R` is the `R` code used to produce them.
The cohort simulated is constituted by 1000 individus with 10000 SNPs. There are 1000 causal SNPs whose effect size is determined by a *rLaplace* distribution. The prevalence is 0.25.

```{r}
fam = "data/sim9d.fam"
bim = "data/sim9d.bim"
bed = "data/sim9d.bed"
dat = read.plink(bed, bim, fam)
```

## Psrank
### Package design

```{r, eval=FALSE}
PSRank(plink="filename",
       frac=.5,
       ranking="Z")
```
Parameters

* `frac`: Fraction of the dataset that's used for training
* `ranking`: What method should be used to rank

This function return a matrix of AUC values. It's a matrix of 100 rows for each replication. The number of SNPs define the number of columns.


### Analysis 
The parameter `f` refers to the fraction of the dataset that's used for training:

```{r}
f = .5
```
#### Z-score
```{r}
res_z = PSRank(dat,
       frac=f,
       ranking="Z")
```

##### Plot
To visualise the results we suggest to do it through a plot. This graph represents, in full line, the AUC average of the 100 replications for each SNP. The AUC variation, in gray, correspond to a confidence interval.

```{r}
m = 10001 # number of SNPs + 1
q = 0.1   # Confidence interval

qAUC_z = colQuantiles(res_z, probs=c(q, .5, 1 - q))

ggplot(tibble(n=0:m,
              median = c(.5, qAUC_z[, 2]),
              lower  = c(.5, qAUC_z[, 1]),
              upper  = c(.5, qAUC_z[, 3])), aes(n)) +
  geom_ribbon(aes(ymin=lower, ymax=upper), fill="gray30", alpha=0.2) +
  geom_line(aes(y=median)) +
  NULL
```

#### I-score
##### Usage

```{r, eval=FALSE}
I_score(genotypes, phenotypes)
```

##### Arguments

* `genotypes`   a $S \times N$ matrix containing 0, 1, 2 or NA;
* `phenotypes` a vector of $S$ values for the phenotype; use numeric values for a quantitative trait and booleans for a case/control association study.

##### PSRank with I-score ranking method
```{r}
res_i = PSRank(dat,
       frac=f,
       ranking="iscore")

```


##### Plot
To visualise the results we suggest to do it through a plot. This graph represents, in full line, the AUC average of the 100 replications for each SNP. The AUC variation, in gray, correspond to a confidence interval.

```{r}
m = 10001 # number of SNPs + 1
q = 0.1   # Confidence interval

qAUC_i = colQuantiles(res_i, probs=c(q, .5, 1 - q))

ggplot(tibble(n=0:m,
              median = c(.5, qAUC_i[, 2]),
              lower  = c(.5, qAUC_i[, 1]),
              upper  = c(.5, qAUC_i[, 3])), aes(n)) +
  geom_ribbon(aes(ymin=lower, ymax=upper), fill="gray30", alpha=0.2) +
  geom_line(aes(y=median)) +
  NULL
```


#### comparison

