---
title: "Vignette for the PSRank package"
author: "Safa-tahar-henni Safia"
date: "29 Oct 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE, comment=NULL)
```

# Introduction

In this vignette, we introduce the main features of the package "PSRank".

Genome-wide association studies (GWASs) are designed to identify genetic susceptibility factors for complex diseases. They are effective approaches for identifying genetic polymorphisms associated with disease risk.  Polygenic risk scores (PRSs), computed from genotyping data, are promising approaches to use genetics in the clinic but they still fall short of the required accuracy to be truly applicable.

The PSRank package provides a user-friendly way to assess the general behaviour of the polygenic risk score as a function of the retained polymorphisms.  This behaviour primarily depends on the genetic architecture of the trait of interest.  It also depends on the ranking metric used.  Traditionally, $p$-values have been used but other statistics could be used.  The $I$ score has been proposed as an alternative to traditional significance testing; it was conceived as a feature selection mechanism predominantly focusing on prediction.

- A learning set where will be estimated associations through linear univariate model. This set of SNPs is going to be sorted. The packages allows to sorte the SNPs according differentes scores. Classically, by P-values or β. The Lo et al., 2015 's article offer an alternative statistic to the $\chi$2, the I-score: designed to recognize strongly predictive variables rather than highly significant. This score can be used to sort a set of SNPs.
- And a test set where will be calculated a cumulative score, called "polygenic risk score" developed by Purcell et al. and on which we will evaluate the predictivity

# Installation

To install the package PSRank and load it into R (version 3.4.2):

```{r}
devtools::install_github("SafiaSafa/PSRank")
library("PSRank")
```

# Package dependencies

The package relies on the following libraries

* [caTools](https://cran.r-project.org/package=caTools)
* [snpStats](https://www.bioconductor.org/packages/release/bioc/html/snpStats.html), a
  [Bioconductor](https://www.bioconductor.org/) package
* [matrixStats](https://cran.r-project.org/package=matrixStats)

Furthermore, in this vignette, we'll use

```{r, message=FALSE, warning=FALSE}
library("tidyverse")
library("snpStats")
library("matrixStats")
```

# Example of analysis

## Loading SNP data

Synthetic data are available in the "data/" directory; they use PLINK's binary format (BED, BIM and FAM extensions). They were produced by simulation.  The data corresponds to 1000 individuals (cases and controls) and 10000 genotyped SNPs. Only 1000 SNPs are causal and effect sizes follow a Laplace distribution. The prevalence is 0.25.

```{r}
fam = system.file("Data/sim9d.fam", package="PSRankp")
bim = system.file("Data/sim9d.bim", package="PSRankp")
bed = system.file("Data/sim9d.bed", package="PSRankp")
dat = read.plink(bed, bim, fam)
```

## PSRank() function

The principal function defined by the package is PSRank()

```{r, eval=FALSE}
PSRank(plink="filename", frac=.75, ranking="Z", method="subsampling", samples=100)
```

Two parameters

* frac — Fraction of the dataset that's used for training
* ranking — Statistic used to rank SNPs by decreasing importance
* method — Cross-validation method (subsampling of kfold)
* samples — Number of replications (subsampling) or $K$ folds

This function return a matrix of AUC values.

## Analysis

Using the $Z$ score as a ranking statistic:

```{r}
Z = PSRank(dat, frac=.50, ranking="Z")
```

The output $(Z_{ij})$ is a matrix containing AUC values; it has as
many columns as SNPs and there's a row for each cross-validation
sample.

## Plot

To visualise the results we suggest to do it through a plot. This graph represents the AUC median of all replications for each SNP (solid line). The spread of the AUC distribution, shaded in gray, is obtained from the quantiles.

```{r}
m = nrow(bimdat$map) # Number of SNPs
q = 0.1              # Uncertainty level

AUC = colQuantiles(Z, probs=c(q, .5, 1 - q))

ggplot(tibble(n=0:m,
              median = c(.5, AUC[, 2]),
              lower  = c(.5, AUC[, 1]),
              upper  = c(.5, AUC[, 3])), aes(n)) +
  geom_ribbon(aes(ymin=lower, ymax=upper), fill="gray30", alpha=0.2) +
  geom_line(aes(y=median)) +
  ggtitle("AUC as a function of the number of SNPs included in the PRS")
```
